doc/deprecation-process.md

```diff
@@ -0,0 +1,272 @@
+```
+
+**If a migration guide is not provided:**
+
+```markdown
+# Microsoft Azure SDK for Rust
+
+> [!CAUTION]
+> This crate has been deprecated and will no longer be maintained after <EOLDate>. This crate will only receive security fixes until <EOLDate>. To receive updates on new features and non-security bug fixes, upgrade to the replacement crate, [azure_new_crate](https://crates.io/crates/azure_new_crate).
+```
+
+**If a replacement crate does not exist:**
+
+```markdown
+# Microsoft Azure SDK for Rust
+
+> [!CAUTION]
+> This crate has been deprecated and will no longer be maintained after <EOLDate>. This crate will only receive security fixes until <EOLDate>.
+```
+
+**If a new service has replaced the service, and existing customers should be directed to the new service's REST API docs/repo:**
+
+```markdown
+# Microsoft Azure SDK for Rust
+
+> [!CAUTION]
+> This crate has been deprecated and will no longer be maintained after <EOLDate>. This crate will only receive security fixes until <EOLDate>. Refer to the samples in the [My New Service repo](https://github.com/microsoft/my-new-service/tree/main) instead.
+
+For additional support, open a new issue in the [Issues](https://github.com/microsoft/my-new-service/issues) section of the My New Service repo.
+```
+
+### CHANGELOG.md and Cargo.toml version
+
+-   Update the version in the crate's `Cargo.toml` file to the next patch version if the crate has had a stable release, or the next pre-release version if the crate has only been in pre-release. For example:
+
+    -   If a stable version WAS NEVER RELEASED and the last released version was 1.0.0-beta.1, the new version should be 1.0.0-beta.2.
+    -   If a stable version HAS BEEN RELEASED and the last released version was 1.2.3-beta.1, the new version should be 1.2.4.
+    -   If the last released version was 1.2.3, the new version should be 1.2.4.
+
+-   In `CHANGELOG.md`, add the new version with the same disclaimer as in the `README.md`, along with a release date. No other changes/features added/breaking changes should be included for this version. For example:
+
+```markdown
+## 1.2.4 (2024-03-31)
+
+### Other Changes
+
+-   This crate has been deprecated and will no longer be maintained after 12-31-2024. This crate will only receive security fixes until 12-31-2024. To receive updates on new features and non-security bug fixes, upgrade to the replacement crate, [azure_new_crate](https://crates.io/crates/azure_new_crate). Refer to the [migration guide](https://aka.ms/azsdk/rust/migrate/new-crate) for guidance on upgrading.
+```
+
+### Source Code Deprecation Attributes
+
+Add `#[deprecated]` attributes to all public APIs in the crate to provide compile-time warnings to users:
+
+```rust
+#[deprecated(since = "1.2.4", note = "This crate has been deprecated. Use azure_new_crate instead.")]
```

Comment:
If they're deprecating a type or member, "Use azure_new_crate instead" isn't a good recommendation. I would be more concrete in recommendations.